## Data Visualization & Communication CA1

- Lecturer: Sam Weiss
- Student Name: Danrlei da Silva Martins
- Student Number: 2020322
- GitHub Repository: https://github.com/danrlei-martins-cct/DVC-CA1

In [1]:
# Import necessary libraries
import numpy as np
import seaborn as sns
import pandas as pd
import matplotlib.pyplot as plt

### 1. Data Loading
- Reading provided dataset from csv file

In [35]:
# Read the CSV file
df = pd.read_csv("fortune500.csv")

### 2. Data Exploration
- Exploring dataset to understand its structure
- Checking and handling missing values

In [36]:
# Rename columns for clarity
df.columns = ['year', 'rank', 'company', 'revenue', 'profit']

In [37]:
# Display the 5 first rows
df.head()

year  rank           company  revenue profit
0  1955     1    General Motors   9823.5    806
1  1955     2       Exxon Mobil   5661.4  584.8
2  1955     3        U.S. Steel   3250.4  195.4
3  1955     4  General Electric   2959.1  212.6
4  1955     5            Esmark   2510.8   19.1

In [38]:
# Display the 5 last rows
df.tail()

year  rank                company  revenue profit
25495  2005   496        Wm. Wrigley Jr.   3648.6    493
25496  2005   497         Peabody Energy   3631.6  175.4
25497  2005   498  Wendy's International   3630.4   57.8
25498  2005   499     Kindred Healthcare   3616.6   70.6
25499  2005   500   Cincinnati Financial   3614.0    584

In [39]:
# Check the data types of the columns
df.dtypes

year         int64
rank         int64
company     object
revenue    float64
profit      object
dtype: object

- Here we see that the profit column data type is an object, which indicates there are non-numeric values there.

In [42]:
# Identify rows with non-numeric values in the 'profit' column
non_numeric_profits = df.profit.str.contains('[^0-9.-]')

# Display the first few rows with non-numeric 'profit' values
df.loc[non_numeric_profits].head()

year  rank                company  revenue profit
228  1955   229                 Norton    135.0   N.A.
290  1955   291        Schlitz Brewing    100.0   N.A.
294  1955   295  Pacific Vegetable Oil     97.9   N.A.
296  1955   297     Liebmann Breweries     96.0   N.A.
352  1955   353     Minneapolis-Moline     77.4   N.A.

- Here we see that the profit column has missing values

In [43]:
# Create a set of unique non-numeric 'profit' values
set(df.profit[non_numeric_profits])

{'N.A.'}

In [44]:
# Calculate the count of rows with non-numeric 'profit' values
len(df.profit[non_numeric_profits])

369

- We discover that we have a total of 369 missing values in the dataset, now we need to decide what to do with them.

In [46]:
# Calculate the percentage of rows with non-numeric 'profit' values relative to the entire dataset
len(df.profit[non_numeric_profits])/len(df)*100

1.4470588235294117

- As we see above, the missing values represent only 1.45 percent.

## References
